File: mobile/src/api/advisoryApi.ts
import axios from 'axios';

const API_BASE_URL = process.env.API_BASE_URL || 'http://localhost:8080/api/v1';

export const advisoryApi = {
  /**
   * Get all advisories for a farmer (for polling fallback).
   */
  async getAll(farmerId: string, authToken: string, unreadOnly: boolean = false) {
    const response = await axios.get(`${API_BASE_URL}/advisories`, {
      params: {
        farmer_id: farmerId,
        unread: unreadOnly
      },
      headers: {
        'Authorization': `Bearer ${authToken}`
      }
    });
    return response.data;
  },

  /**
   * Mark an advisory as read.
   */
  async markAsRead(advisoryId: string, authToken: string) {
    const response = await axios.patch(
      `${API_BASE_URL}/advisories/${advisoryId}/read`,
      {},
      {
        headers: {
          'Authorization': `Bearer ${authToken}`
        }
      }
    );
    return response.data;
  },

  /**
   * Get AI advisory (chat endpoint).
   */
  async getAdvisory(request: {
    farmer_id: string;
    crop_type: string;
    stage: string;
    language: string;
    question: string;
  }, authToken: string) {
    const response = await axios.post(
      `${API_BASE_URL}/advisory/chat`,
      request,
      {
        headers: {
          'Authorization': `Bearer ${authToken}`
        }
      }
    );
    return response.data;
  }
};

File: mobileV2/app/src/main/java/com/krishimitra/mobilev2/data/api/AdvisoryApi.kt
package com.krishimitra.mobilev2.data.api

import com.krishimitra.mobilev2.data.model.AdvisoryChatResponse
import retrofit2.Call
import retrofit2.http.*

/**
 * Retrofit interface for Advisory API endpoints.
 * Equivalent to Spring Boot @RestController for advisory chat.
 */
interface AdvisoryApi {
    
    /**
     * Get advisory from AI based on question.
     * POST /api/v1/advisory/chat
     */
    @POST("advisory/chat")
    fun getAdvisory(@Body request: AdvisoryChatRequest): Call<AdvisoryChatResponse>

    /**
     * Get list of advisories for a farmer.
     */
    @GET("advisories")
    fun getAdvisories(
        @Query("farmer_id") farmerId: String,
        @Query("unread") unread: Boolean = false
    ): Call<AdvisoryListResponse>

    /**
     * Mark an advisory as read.
     */
    @PATCH("advisories/{id}/read")
    fun markAsRead(@Path("id") id: String): Call<Any>
}

// Request/Response DTOs
data class AdvisoryListResponse(
    val advisories: List<Map<String, Any>>,
    val unread_count: Int
)

data class AdvisoryChatRequest(
    val farmer_id: String,
    val crop_type: String,
    val stage: String,
    val language: String,
    val question: String
)
